uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/CardDetailsCheckoutScreen.kt

```diff
@@ -277,31 +276,34 @@ private fun CvvField(
     viewModel: CardDetailsCheckoutViewModel,
 ) {
     val scrollOffset = with(LocalDensity.current) {
-        if (state.isEmailInputFieldRequired && state.isPostalCodeFieldRequired) {
+        if (state.isBillingCountryFieldRequired && state.isPostalCodeFieldRequired) {
             FIFTH_FIELD_OFF_SET_DP.dp.toPx()
-        } else if (state.isPostalCodeFieldRequired || state.isEmailInputFieldRequired) {
+        } else if (state.isBillingCountryFieldRequired || state.isPostalCodeFieldRequired) {
             FORTH_FIELD_OFF_SET_DP.dp.toPx()
         } else {
             THIRD_FIELD_OFF_SET_DP.dp.toPx()
         }
     }
-
+    var focusedTextKey by remember { mutableStateOf(false) }
     CvvInputField(
-        modifier = Modifier
-            .onFocusChanged {
-                if (it.hasFocus) {
-                    coroutineScope.launch {
-                        delay(300)
-                        scrollState.animateScrollTo(
-                            scrollToPosition.roundToInt() + scrollOffset.roundToInt(),
-                        )
-                    }
+        modifier = Modifier.onFocusChanged {
+            focusedTextKey = if (it.isFocused) {
+                coroutineScope.launch {
+                    delay(300)
+                    scrollState.animateScrollTo(
+                        scrollToPosition.roundToInt() + scrollOffset.roundToInt(),
+                    )
                 }
-                viewModel.validateCvv(
-                    state.cvvInputFieldState.value,
-                    it.isFocused,
-                )
-            },
+                true
+            } else {
+                if (focusedTextKey) {
+                    viewModel.validateCvv(
+                        state.cvvInputFieldState.value,
+                    )
+                }
+                false
+            }
```

Comment:
I see that this code is being repeated a few times, can we make a function to reuse it?